### ISSUE TITLE
DNS geo mapping in ElasticSearch

### ISSUE THREAD

**Issue by luckyb56** [StamusNetworks/SELKS#10]
Logstash takes src_ip or dest_ip for Geo location which is fine for other protocols, but it should take [dns.rdata] for DNS as scr_ip could be local DNS IP.

**Comment by pevma** [comment]
@luckyb56  - the IP(v4/6) is present in dns.rdata for A or AAAA records, correct?. Just making sure I we are not missing something before adjusting the conf.

**Comment by luckyb56** [comment]
That's correct. An example from my data having AAAA record:
"dns": {
      "type": "answer",
      "id": 33667,
      "rrname": "dns.msftncsi.com",
      "rrtype": "AAAA",
      "ttl": 3321,
      "rdata": "fd3e:4f5a:5b81:0000:0000:0000:0000:0001"
    }

**Comment by pevma** [comment]
Just wondering  - if you have already done it - is there any performance hit we should worry about?

**Comment by luckyb56** [comment]
Unfortunately I ignore IPv6 in post process. When you say changing the conf, you mean logstash conf, right? I can do a test run for you. I'll know if my intake has reduced.

**Comment by pevma** [comment]
Yes - I meant changing logstash.conf. Can you please feedback (thank you for suggesting) if there is any perf hit we should cnsider?

**Comment by luckyb56** [comment]
Sure. If you have made the changes, please share it with me. I'll update my conf and let you know in a day or two.

**Comment by pevma** [comment]
ok thanks - I will let you know as soon as I have something.

**Comment by pevma** [comment]
I have a working logstash.conf that you can try - was wondering how to get it to you.....
